config/config.go

```diff
@@ -79,8 +78,11 @@ func NewConfigHandler(pollInterval time.Duration, httpClient *http.Client, user
 		apiClient:               common.NewWebClient(httpClient),
 		preferredServerLocation: atomic.Value{}, // initially, no preference
 	}
+	// Store an empty preferred location to avoid nil pointer dereference
```

Comment:
this is not good. setting the value to empty Preferred Region will break the API call, since it assumes that only 'nil' means No Preference and it will try to search for empty city and country

Reply:
Does the previous code also specify an empty server and country? The `preferredServerLocation := &serverLocation{}` below?

```go
locationPtr := ch.preferredServerLocation.Load()
preferredServerLocation := &serverLocation{}
if locationPtr != nil {
    preferredServerLocation = locationPtr.(*serverLocation)
}
```

Reply:
@myleshorton hah. no, it's me being an idiot. I'm lecturing you about something I didn't do properly myself 😆 
it was supposed to be
```
var preferredServerLocation *PreferredRegion
```
instead of the empty assignment